explorer/pkg/xurl/xurl_test.go

```diff
@@ -0,0 +1,192 @@
+package xurl
+
+import (
+	"net/url"
+	"testing"
+
+	"github.com/ignite/cli/v28/ignite/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+func TestParse(t *testing.T) {
+	cases := []struct {
+		name string
+		addr string
+		want *url.URL
+		err  error
+	}{
+		{
+			name: "http",
+			addr: "http://localhost",
+			want: &url.URL{Host: "localhost:80", Scheme: schemeHTTP},
+		},
+		{
+			name: "https",
+			addr: "https://localhost",
+			want: &url.URL{Host: "localhost:443", Scheme: schemeHTTPS},
+		},
+		{
+			name: "custom",
+			addr: "http://localhost:4000",
+			want: &url.URL{Host: "localhost:4000", Scheme: schemeHTTP},
+		},
+		{
+			name: "custom ssl",
+			addr: "https://localhost:4005",
+			want: &url.URL{Host: "localhost:4005", Scheme: schemeHTTPS},
+		},
+		{
+			name: "no schema and port",
+			addr: "localhost",
+			want: &url.URL{Host: "localhost:80", Scheme: schemeHTTP},
+		},
+		{
+			name: "no schema",
+			addr: "localhost:80",
+			want: &url.URL{Host: "localhost:80"},
+		},
+		{
+			name: "invalid address",
+			addr: "://.e",
+			err:  errors.New("parse \"://.e\": missing protocol scheme"),
+		},
+	}
+
+	for _, tt := range cases {
+		t.Run(tt.name, func(t *testing.T) {
+			addr, err := Parse(tt.addr)
+			if tt.err != nil {
+				require.Error(t, err)
+				require.Equal(t, tt.err.Error(), err.Error())
+				return
+			}
+			require.NoError(t, err)
+			require.EqualValues(t, tt.want, addr)
+		})
+	}
+}
+
+func Test_addressPort(t *testing.T) {
```

Comment:
Maybe we could omit the private function tests by making sure that these cases are included in the public function tests